Wikipedia/View Controllers/WebView/WebViewController.m

```diff
@@ -257,6 +261,33 @@ - (void)viewDidLoad {
     self.webView.scrollView.layer.anchorPoint = CGPointMake((isRTL ? 1.0 : 0.0), 0.0);
 
     [self tocUpdateViewLayout];
+
+    [self.KVOControllerNonRetaining observe:self.webView.scrollView keyPath:WMF_SAFE_KEYPATH(UIScrollView.new, contentSize) options:NSKeyValueObservingOptionNew block:^(WebViewController* observer, id object, NSDictionary* change) {
```

Comment:
seems like the section header VC's abstraction is leaking into the webVC. I think you'll cover this by giving the section VC a (weak) reference to the webview it's reacting to.